### PR TITLE
Allowing name input for New MultiModels projects through w2prompt (W2UI 1.5) dialogs

### PR DESCRIPTION
**Pull request by idhugo**

### CODE REVIEW

File: src/init.ts [hunk 1]
```diff
@@ -207,10 +207,33 @@ menuHandler.createMultiModel = (path) => {
...
+            height      : 200,
+            callBack    : function (value : String) {
+
+                let content = fs.readFileSync(path, "UTF-8");
+                try {
+                let mmPath = <string>project.createMultiModel(value, content);
+                menuHandler.openMultiModel(mmPath);
+                } catch (error){
+                    // TODO: Validate name! Distinguish names... Maybe with the help of the recent w2prompt :with
+                    w2alert('MM Project '+  value + ' already exists! Choose a different name.',"Error");
```
Comment (CThuleHansen) [inline src/init.ts]: I suggest changing the naming from "MM Project" to "Multi-Model"

File: src/init.ts [hunk 2]
```diff
@@ -230,11 +253,34 @@ menuHandler.createMultiModelPlain = () => {
...
+            cancel_text : 'Cancel',
+            width       : 400,
+            height      : 200,
+            callBack    : function (value : String) {
+                try {
+                let mmPath = <string>project.createMultiModel(value, "{}");
+                menuHandler.openMultiModel(mmPath);
+                } catch (error){
+                    // TODO: Validate name! Distinguish names... Maybe with the help of the recent w2prompt :with
+                    w2alert('MM Project '+  value + ' already exists! Choose a different name.',"Error");
```
Comment (CThuleHansen) [inline src/init.ts]: I suggest changing the naming from "MM Project" to "Multi-Model"

File: src/proj/Project.ts [hunk 3]
```diff
@@ -108,6 +108,8 @@ export class Project implements IProject {
     public createMultiModel(name: String, jsonContent: String): String {
         let path = Path.normalize(this.rootPath + "/" + Project.PATH_MULTI_MODELS + "/" + name);
 
+        if(fs.existsSync(path)) throw new Error('MM Project '+  name + ' already exists!');
```
Comment (CThuleHansen) [inline src/proj/Project.ts]: I suggest changing the naming from "MM Project" to "Multi-Model"

File: src/init.ts [hunk 4]
```diff
@@ -207,10 +207,33 @@ menuHandler.createMultiModel = (path) => {
...
-        let mmPath = <string>project.createMultiModel(`mm-${name} (${Math.floor(Math.random() * 100)})`, content);
-        menuHandler.openMultiModel(mmPath);
+        let ivname  = `mm-${name}`;
+        
+
+        w2prompt({
+            label       : 'Name',
+            value       : ivname,
+            attrs       : 'style="width: 200px"',
+            title       : 'New MM Project',
```
Comment (CThuleHansen) [inline src/init.ts]: I suggest changing the naming from "MM Project" to "Multi-Model"

**Comment by CThuleHansen** [comment]
Hey @idhugo.
I will launch the application using the branch during the weekend.
I have left a few comments on the naming

**Comment by CThuleHansen** [comment]
Hey @idhugo.
I will launch the application using the branch during the weekend.
I have left a few comments on the naming

File: src/proj/Project.ts (hunk 3 above)
Comment (idhugo) [inline src/proj/Project.ts]: Nice suggestion! Done...

File: src/init.ts [hunk 5]
```diff
@@ -218,12 +218,32 @@ menuHandler.createMultiModel = (path) => {
...
+                let content = fs.readFileSync(path, "UTF-8");
+                try {
+                let mmPath = <string>project.createMultiModel(value, content);
+                menuHandler.openMultiModel(mmPath);
+                //Create the trace 
+                let message = TraceMessager.submitSysMLToMultiModelMessage(mmPath,path);
+                //console.log("RootMessage: " + JSON.stringify(message));    
+                } catch (error){
+                    // TODO: Validate name! Distinguish names... Maybe with the help of the recent w2prompt :with
+                    w2alert('Multi-Model Project '+  value + ' already exists! Choose a different name.',"Error");
```
Comment (CThuleHansen) [inline src/init.ts]: just "Multi-Model". No need for "Project"

File: src/init.ts [hunk 6]
```diff
@@ -255,11 +275,34 @@ menuHandler.createMultiModelPlain = () => {
...
+            cancel_text : 'Cancel',
+            width       : 400,
+            height      : 200,
+            callBack    : function (value : String) {
+                try {
+                let mmPath = <string>project.createMultiModel(value, "{}");
+                menuHandler.openMultiModel(mmPath);
+                } catch (error){
+                    // TODO: Validate name! Distinguish names... Maybe with the help of the recent w2prompt :with
+                    w2alert('Multi-Model Project '+  value + ' already exists! Choose a different name.',"Error");
```
Comment (CThuleHansen) [inline src/init.ts]: I do not think you need "Project", just "Multi-Model"

Comment (CThuleHansen) [inline src/init.ts]: Relaunch the prompt if the name already exists?

File: src/init.ts (hunk 5 above)
Comment (CThuleHansen) [inline src/init.ts]: Relaunch the prompt if the name already exists?

File: src/init.ts [hunk 7]
```diff
@@ -218,12 +218,32 @@ menuHandler.createMultiModel = (path) => {
...
-        //Create the trace 
-        let message = TraceMessager.submitSysMLToMultiModelMessage(mmPath,path);
-        //console.log("RootMessage: " + JSON.stringify(message));    
+        let ivname  = `mm-${name}`;
+
+        w2prompt({
+            label       : 'Name',
+            value       : ivname,
+            attrs       : 'style="width: 200px"',
+            title       : 'New Multi-Model Project',
```
Comment (CThuleHansen) [inline src/init.ts]: just "New Multi-Model". No need for "Project"

File: src/init.ts [hunk 8]
```diff
@@ -255,11 +275,34 @@ menuHandler.createMultiModelPlain = () => {
...
-        menuHandler.openMultiModel(mmPath);
+
+        let ivname  = `mm-new`;
+        
+
+        w2prompt({
+            label       : 'Name',
+            value       : ivname,
+            attrs       : 'style="width: 200px"',
+            title       : 'New Multi-Model Project',
```
Comment (CThuleHansen) [inline src/init.ts]: just "New Multi-Model". No need for "Project"

File: src/proj/Project.ts [hunk 9]
```diff
@@ -106,6 +106,8 @@ export class Project implements IProject {
...
```
Comment (CThuleHansen) [inline src/proj/Project.ts]: just "Multi-Model". No need for "Project"

File: src/init.ts [hunk 10]
```diff
@@ -230,11 +253,34 @@ menuHandler.createMultiModelPlain = () => {
...
-        menuHandler.openMultiModel(mmPath);
+
+        let ivname  = `mm-new`;
+        
+
+        w2prompt({
+            label       : 'Name',
+            value       : ivname,
+            attrs       : 'style="width: 200px"',
+            title       : 'New MM Project',
```
Comment (CThuleHansen) [inline src/init.ts]: just "New Multi-Model". No need for "Project"

File: src/proj/Project.ts (hunk 3 above)
Comment (CThuleHansen) [inline src/proj/Project.ts]: A few things:

- I think you should remove "Project" from "Multi-Model Project".
- If you press cancel when creating a new Multi-Model, then it creates a file called null
- If the name already exists, then spawn the prompt again with the possibility of naming the file.

File: src/init.ts (hunk 5 above)
Comment (CThuleHansen) [inline src/init.ts]: Relaunch the prompt if name already exists

**Comment by CThuleHansen** [comment]
A few things:

- I think you should remove "Project" from "Multi-Model Project".
- If you press cancel when creating a new Multi-Model, then it creates a file called null
- If the name already exists, then spawn the prompt again with the possibility of naming the file.

**Comment by idhugo** [comment]
@CThuleHansen  Please check new changes!

File: src/init.ts [hunk 11]
```diff
@@ -212,18 +214,39 @@ menuHandler.openFmu = () => {
...
+                try {
+                    if (!value) {return;}
+                    let mmPath = <string>project.createMultiModel(value, content);
+                    menuHandler.openMultiModel(mmPath);
+                } catch (error){
+                    menuHandler.createMultiModel(path,'Multi-Model "'+  value + '" already exists! Choose a different name.');
+                    return;
+                }
+                //Create the trace 
+                let message = TraceMessager.submitSysMLToMultiModelMessage(mmPath,path);
```
Comment (CThuleHansen) [inline src/init.ts]: I am getting the compiler error: 
src\init.ts(246,76): error TS2304: Cannot find name 'mmPath'.

File: src/proj/Project.ts (hunk 9 above)
Comment (CThuleHansen) [inline src/proj/Project.ts]: This has still not been fixed.
It should be "Multi-Model", not "MM Project"

Comment (idhugo) [inline src/proj/Project.ts]: That is an Error class message  that is never passed to the user... But if that pleases you I already changed it.

Comment (CThuleHansen) [inline src/proj/Project.ts]: Ahh ok. You could just have mentioned in the previous comments I made, then I would not have kept suggesting it.
Cool

File: src/init.ts (hunk 11 above)
Comment (idhugo) [inline src/init.ts]: Solved. Sorry... I missed that in the huge amount of typing warnings my gulp issues.

**Comment by CThuleHansen** [comment]
@idhugo looks good! Ready to merge from my point of view
@lausdahl @ldcouto looks ok?
@ldcouto will you do the merge? So we do not fuck up related to any release date :)

**Pull request merged**